src/unistd.rs

```diff
@@ -2455,6 +2458,79 @@ impl From<*mut libc::passwd> for User {
     }
 }
 
+impl User {
+    /// Get a user by UID.
+    ///
+    /// Internally, this function calls
+    /// [getpwuid_r(3)](http://pubs.opengroup.org/onlinepubs/9699919799/functions/getpwuid_r.html)
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use nix::unistd::{Uid, User};
+    /// // Returns an Option<Result<User>>, thus the double unwrap.
+    /// let res = User::from_uid(Uid::from_raw(0), Some(2048)).unwrap().unwrap();
+    /// assert!(res.name == "root");
+    /// ```
+    pub fn from_uid(uid: Uid, bufsize: Option<usize>) -> Option<Result<Self>> {
+        let mut cbuf = vec![0 as c_char; bufsize.unwrap_or(PWGRP_BUFSIZE)];
```

Comment:
I tried and it fails ... ideas how to avoid initializing it?

Reply:
Like this:
```rust
let mut cbuf  = Vec::with_capacity(...);
libc::getpwuid_r(..., cbuf.as_mut_ptr(), cbuf.capacity(), ...);
```
Ordinarily you would follow that up with a `cbuf.set_len`.  But in this case you don't need to, because no Rust code ever examines its contents.

Reply:
Done.